src/Collapse/include/gudhi/Flag_complex_edge_collapser.h

```diff
@@ -0,0 +1,364 @@
+  {
+    GUDHI_CHECK((u != v), std::invalid_argument("Flag_complex_edge_collapser::insert_new_edge with u == v"));
+    // The edge must not be added before, it should be a new edge.
+    IVertex rw_u = insert_vertex(u);
+    IVertex rw_v = insert_vertex(v);
+#ifdef DEBUG_TRACES
+    std::cout << "Inserting the edge " << u <<", " << v << std::endl;
+#endif  // DEBUG_TRACES
+    sparse_row_adjacency_matrix_[rw_u].insert(rw_v) = idx;
+    sparse_row_adjacency_matrix_[rw_v].insert(rw_u) = idx;
+    return std::minmax(rw_u, rw_v);
+  }
+
+ public:
+  /** \brief Flag_complex_edge_collapser constructor from a range of filtered edges.
+   *
+   * @param[in] begin Iterator on the first element of a filtered edges range, aka. `std::begin`. Filtered edges must
+   * be in `Flag_complex_edge_collapser::Filtered_edge`.
+   *
+   * @param[in] end Iterator on the final element of a filtered edges range, aka. `std::end`. Filtered edges must be
+   * in `Flag_complex_edge_collapser::Filtered_edge`.
+   *
+   * There is no need the range to be sorted, as it will be performed in
+   * `Flag_complex_edge_collapser::process_edges`.
+   */
+  template<typename Filtered_edge_iterator>
+  Flag_complex_edge_collapser(Filtered_edge_iterator begin, Filtered_edge_iterator end)
+  : f_edge_vector_(begin, end) { }
+
+  /** \brief Inserts all edges given by a OneSkeletonGraph into a vector of
+   * `Flag_complex_edge_collapser::Filtered_edge`.
+   * OneSkeletonGraph must be a model of
+   * <a href="http://www.boost.org/doc/libs/1_65_1/libs/graph/doc/EdgeListGraph.html">boost::EdgeListGraph</a>
+   * and <a href="http://www.boost.org/doc/libs/1_65_1/libs/graph/doc/PropertyGraph.html">boost::PropertyGraph</a>.
+   *
+   * The edge filtration value is accessible through the property tag
+   * edge_filtration_t.
+   *
+   * boost::graph_traits<OneSkeletonGraph>::vertex_descriptor
+   *                                    must be Vertex_handle.
+   * boost::graph_traits<OneSkeletonGraph>::directed_category
+   *                                    can be directed_tag (the fastest, the least RAM use), undirected_tag or even
+   *                                    bidirected_tag.
+   *
+   * If an edge appears with multiplicity, the function will arbitrarily pick one representative to read the filtration
+   * value.
+   * 
+   * `Gudhi::Proximity_graph<Flag_complex_edge_collapser>` is a good candidate for OneSkeletonGraph.
+   */
+  template<class OneSkeletonGraph>
+  Flag_complex_edge_collapser(const OneSkeletonGraph& one_skeleton_graph) {
+    // Insert all edges
+    for (auto edge_it = boost::edges(one_skeleton_graph);
+         edge_it.first != edge_it.second; ++edge_it.first) {
+      auto edge = *(edge_it.first);
+      Vertex_handle u = source(edge, one_skeleton_graph);
+      Vertex_handle v = target(edge, one_skeleton_graph);
+      f_edge_vector_.emplace_back(Filtered_edge(u, v, boost::get(Gudhi::edge_filtration_t(), one_skeleton_graph, edge)));
```

Comment:
I think we are not supposed to write `boost::` in front for `edges` and `get`, according to the boost EdgeListGraph and PropertyGraph concepts.

Reply:
Done on ddd12a6c